pkg/resolution/pm/pip/job.go

```diff
@@ -112,6 +105,44 @@ func (j *Job) Run() {
 	if err != nil {
 		j.Errors().Critical(err)
 	}
+
+	if j.install {
+		j.SendStatus("removing virtualenv")
+		err = j.pipCleaner.RemoveAll(j.venvPath)
+		if err != nil {
+			j.Errors().Critical(err)
+		}
+	}
+}
+
+func (j *Job) generateLockContent() (catCmdOutput, listCmdOutput, ShowCmdOutput []byte, err error) {
+
+	j.SendStatus("running cat command")
+	catCmdOutput, err = j.runCatCmd()
+	if err != nil {
+		j.Errors().Critical(err)
+
+		return nil, nil, nil, err
+	}
+
+	j.SendStatus("running list command")
+	listCmdOutput, err = j.runListCmd()
+	if err != nil {
+		j.Errors().Critical(err)
+
+		return nil, nil, nil, err
+	}
+
+	j.SendStatus("running show command")
+	installedPackages := j.parsePipList(string(listCmdOutput))
+	ShowCmdOutput, err = j.runShowCmd(installedPackages)
+	if err != nil {
+		j.Errors().Critical(err)
+
+		return nil, nil, nil, err
+	}
+
+	return catCmdOutput, listCmdOutput, ShowCmdOutput, nil
```

Comment:
Might be nice to combine the strings to only return one output string, since the return only is used combined later on as well. Looks much better than having multiple returns.